libutil/dr_frontend_unix.c

```diff
@@ -166,15 +168,18 @@ drfront_searchenv(const char *fname, const char *env_var, OUT char *full_path,
         NULL_TERMINATE_BUFFER(tmp);
         /* realpath checks for existence too. */
         if (realpath(tmp, realpath_buf) != NULL) {
-            status_check = drfront_access(realpath_buf, DRFRONT_EXIST, &access_ret);
+            status_check = drfront_access(realpath_buf, DRFRONT_EXEC, &access_ret);
             if (status_check != DRFRONT_SUCCESS) {
                 *ret = false;
                 return status_check;
             } else if (access_ret == true) {
-                *ret = true;
-                snprintf(full_path, full_path_size, "%s", realpath_buf);
-                full_path[full_path_size - 1] = '\0';
-                return DRFRONT_SUCCESS;
+                r = stat(realpath_buf, &st);
+                if (r != -1 && !S_ISDIR(st.st_mode)) {
+                    *ret = true;
+                    snprintf(full_path, full_path_size, "%s", realpath_buf);
+                    full_path[full_path_size - 1] = '\0';
+                    return DRFRONT_SUCCESS;
+                }
```

Comment:
Could we include a test in suite/tests/libutil/frontend_test.c?  It already creates a dir: could add that to PATH with setenv and then test whether it shows up in searchenv?

Reply:
I added a test case that tests this. It's a bit convoluted though because the current directory is handled differently.
I'm not sure if it could be written in a bit more elegant way.

Reply:
Looks reasonable.